File: src/semana-2/Guess-Number-Higher-or-Lower.cpp
#include <bits/stdc++.h>
using namespace std;
// https://leetcode.com/problems/guess-number-higher-or-lower/
// 374. Guess Number Higher or Lower

/** 
 * Forward declaration of guess API.
 * @param  num   your guess
 * @return 	     -1 if num is lower than the guess number
 *			      1 if num is higher than the guess number
 *               otherwise return 0
 * int guess(int num);
 */
int pick = 25;
int guess(int num){
    if(num > pick) return -1;
    if(num < pick) return 1;
    return 0;
}
class Solution {
public:
    int guessNumber(long long int n) {
        long long int init = 1, end = n, mid = (n+1)/2;
        while(init <= end){
            if(guess(mid) == -1){
                end = mid - 1;
                mid = (init + end) / 2;
            }else if(guess(mid) == 1){
                init = mid + 1;
                mid = (init + end) / 2;
            }else{
                return mid;
            }
            
        }
        return 2;
    }
};
int main(){
    Solution s;
    cout << s.guessNumber(100) << endl;
    return 0;
}

File: src/semana-2/JavaScript/Guess-Number-Higher-or-Lower.js
/** 
 * Forward declaration of guess API.
 * @param {number} num   your guess
 * @return 	     -1 if num is higher than the picked number
 *			      1 if num is lower than the picked number
 *               otherwise return 0
 * var guess = function(num) {}
 */

/**
 * @param {number} n
 * @return {number}
 * https://leetcode.com/problems/guess-number-higher-or-lower/
 */

const pickedNumber = 25;
const n = 100;

var guess = (num) => {
  if (num > pickedNumber) {
    return -1;
  } else if (num < pickedNumber) {
    return 1;
  } else {
    return 0;
  }
}

var guessNumber = (n) => {
  var init = 1, end = n;
  var mid = 0;

  while (guess(mid) !== 0) {
    mid = Math.floor((end + init) / 2);

    if (guess(mid) === -1) {
      end = mid - 1;
    } else {
      init = mid + 1;
    }
  }

  return mid;
};

console.log(guessNumber(n));